### ISSUE TITLE
LocalTopicMatcher._embed_texts never calls tokenizer or model — prefilter always fails with "BertTokenizer has no attribute items"

### ISSUE THREAD

**Issue by amgrbi96** [CarinaSchoppe/PISMA-Literature-Review-Pipeline-Automation-Tool#42]
### Summary

`LocalTopicMatcher._embed_texts()` in `analysis/topic_prefilter.py` never actually calls the tokenizer or model. Instead it assigns the bare objects to local variables and tries to call `.items()` / forward pass on them, causing every paper's topic prefilter to fail with:

```
WARNING | analysis.topic_prefilter | Local topic prefiltering failed for 'Paper Title': BertTokenizer has no attribute items
```

The pipeline silently falls back to heuristic-only screening, so the local semantic topic prefilter is completely non-functional.

### Affected code

`analysis/topic_prefilter.py`, method `_embed_texts` (currently lines ~518–532):

```python
def _embed_texts(self) -> Any:
    assert self._tokenizer is not None
    assert self._model is not None
    assert self._torch is not None

    encoded = self._tokenizer                                          # ← assigns tokenizer object, not tokenized output
    encoded = {key: value.to(self._device) for key, value in encoded.items()}  # ← calls .items() on tokenizer → fails
    with self._torch.no_grad():
        output = self._model                                           # ← assigns model object, not forward pass output
    token_embeddings = output.last_hidden_state                       # ← would fail if reached
```

Two problems:
1. **Line `encoded = self._tokenizer`** — assigns the tokenizer *object* instead of calling it to encode text. Should be `self._tokenizer(texts, padding=True, truncation=True, return_tensors="pt")`.
2. **Line `output = self._model`** — assigns the model *object* instead of calling the forward pass. Should be `self._model(**encoded)`.

Additionally, the method signature `def _embed_texts(self) -> Any` takes no text argument, so even if the tokenizer were called, there would be nothing to encode.

### Impact

- **Every** paper screened with `topic_prefilter_enabled=True` hits the exception path
- The `except` block in `score_paper` catches the error, logs a warning, and returns `None`
- Downstream code treats `None` as "prefilter unavailable" rather than "prefilter failed"
- The local MiniLM embedding model is loaded into memory but never actually used
- `score_paper` callers in `ai_screener.py` and `relevance_scoring.py` that depend on `.similarity`, `.classification`, `.should_exclude` etc. never get topic-match data

### Reproduction

```bash
py -3 main.py --config-file tests/fixtures/offline_config.json
```

Observe WARNING lines like:
```
Local topic prefiltering failed for 'Large language models for systematic review screening': BertTokenizer has no attribute items
```

### Expected behavior

`_embed_texts` should:
1. Accept a list of text strings as input
2. Call `self._tokenizer(texts, padding=True, truncation=True, return_tensors="pt")` to tokenize
3. Call `self._model(**encoded)` to run the forward pass
4. Return mean-pooled, L2-normalized embeddings

### Suggested fix

```python
def _embed_texts(self, texts: list[str]) -> Any:
    assert self._tokenizer is not None
    assert self._model is not None
    assert self._torch is not None

    encoded = self._tokenizer(texts, padding=True, truncation=True, return_tensors="pt")
    encoded = {key: value.to(self._device) for key, value in encoded.items()}
    with self._torch.no_grad():
        output = self._model(**encoded)
    token_embeddings = output.last_hidden_state
    attention_mask = encoded["attention_mask"].unsqueeze(-1).expand(token_embeddings.size()).float()
    pooled = (token_embeddings * attention_mask).sum(dim=1) / attention_mask.sum(dim=1).clamp(min=1e-9)
    return self._torch.nn.functional.normalize(pooled, p=2, dim=1)
```

And update the two call sites:

```python
# In score_paper (line ~229):
review_embedding, paper_embedding = self._embed_texts([self._review_text, paper_text])

# In _embedding_for_text (line ~488):
embeddings = self._embed_texts([normalized])
```

### Environment

- Python 3.14.4, macOS (Homebrew)
- `transformers==5.9.0`
- `torch==2.12.0`
- Model: `BAAI/bge-small-en-v1.5`
